specification/programmableconnectivity/resource-manager/Microsoft.ProgrammableConnectivity/preview/2023-11-01-preview/OpenApiGateways.json

```diff
@@ -0,0 +1,745 @@
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "OpenApiGatewayListResult": {
+      "type": "object",
+      "description": "The response of a OpenApiGateway list operation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The OpenApiGateway items on this page",
+          "items": {
+            "$ref": "#/definitions/OpenApiGateway"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "OpenApiGatewayProperties": {
+      "type": "object",
+      "description": "APC Gateway properties",
+      "properties": {
+        "gatewayId": {
+          "type": "string",
+          "description": "APC Gateway resource ID linked to this Open API Gateway."
+        },
+        "openApiGatewayOfferingId": {
+          "type": "string",
+          "description": "Reference to the Open API Gateway Offering Resource ID"
+        },
+        "features": {
+          "type": "array",
+          "description": "List of features the user has selected for the current operator in the country provided in the Offering.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "accountType": {
+          "$ref": "#/definitions/AccountType",
+          "description": "Type of the account the user has with the Operator's Network API infrastructure. AzureManaged | UserManaged"
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The status of the last operation.",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "gatewayId",
+        "openApiGatewayOfferingId",
+        "features",
+        "accountType"
+      ]
+    },
+    "OpenApiGatewayUpdate": {
+      "type": "object",
+      "description": "The type used for update operations of the OpenApiGateway.",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "description": "Resource tags.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/OpenApiGatewayUpdateProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "OpenApiGatewayUpdateProperties": {
+      "type": "object",
+      "description": "The updatable properties of the OpenApiGateway.",
+      "properties": {
+        "gatewayId": {
```

Comment:
Is it an azure resource ID? If so, please add "format": "arm-id" and "x-ms-arm-id-details".

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details